tests/homePageTests.spec.js

```diff
@@ -0,0 +1,71 @@
+import { test } from '@playwright/test'
+import HomePage from '../PageObjects/homePageFns'
+import Verify from '../Utils/Assertions/verify'
+import Click from '../Utils/Interaction/click'
+import Type from '../Utils/Interaction/type'
+import PageSels from '../Utils/Selectors/pageSels'
+import TestData from '../TestData/products.json'
+import Constant from '../Utils/Constants/homePageConstants'
+import OrderTypeData from '../TestData/orderTypeDetails.json'
+
+
+const { menuBtn, menuSearchBar, cartCountIcon, searchResultTitle } = PageSels.homePage
+const { item1, item2 } = TestData
+const { pickUp, delivery } = Constant.orderTypes
+const { messageAfterItemAddition } = Constant.toastMessages
+
+let click, verify, homePage, urlCheck, type, baseURL
+test.describe('Should execute KFC India test scenarios- Home page- Menu', () => {
+    test.beforeEach('It should navigate to home page before each test block', async ({ page }) => {
+        homePage = new HomePage(page)
+        verify = new Verify(page)
+        click = new Click(page)
+        type = new Type(page)
+        baseURL = page.context()._options.baseURL
+
+        //navigating to home page and verifying the URL
+        await homePage.navigateToHomePage('/')
+        urlCheck = await verify.theUrl()
```

Comment:
best way to integrate is to have a callback when navigated to specific page and handle the url verification in call back